pkg/gofr/datasource/pubsub/nats/pubsub_wrapper.go

```diff
@@ -0,0 +1,73 @@
+package nats
+
+import (
+	"context"
+
+	"gofr.dev/pkg/gofr/datasource"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+// PubSubWrapper adapts Client to pubsub.JetStreamClient.
+type PubSubWrapper struct {
+	Client *Client
+}
+
+// Publish publishes a message to a topic.
+func (w *PubSubWrapper) Publish(ctx context.Context, topic string, message []byte) error {
+	return w.Client.Publish(ctx, topic, message)
+}
+
+// Subscribe subscribes to a topic and returns a single message.
+func (w *PubSubWrapper) Subscribe(ctx context.Context, topic string) (*pubsub.Message, error) {
+	return w.Client.Subscribe(ctx, topic)
+}
+
+// CreateTopic creates a new topic (stream) in NATS JetStream.
+func (w *PubSubWrapper) CreateTopic(ctx context.Context, name string) error {
+	return w.Client.CreateTopic(ctx, name)
+}
+
+// DeleteTopic deletes a topic (stream) in NATS JetStream.
+func (w *PubSubWrapper) DeleteTopic(ctx context.Context, name string) error {
+	return w.Client.DeleteTopic(ctx, name)
+}
+
+// Close closes the Client.
+func (w *PubSubWrapper) Close() error {
+	ctx := context.Background()
+	return w.Client.Close(ctx)
+}
+
+// Health returns the health status of the Client.
+func (w *PubSubWrapper) Health() datasource.Health {
+	return w.Client.Health()
+}
+
+// Connect establishes a connection to NATS.
+func (w *PubSubWrapper) Connect() {
```

Comment:
I would expect to return an error, an not simply log it

Reply:
You are correct on that but currently this method is satisfying the provider interface:

```go
type provider interface {
	// UseLogger sets the logger for the Cassandra client.
	UseLogger(logger any)

	// UseMetrics sets the metrics for the Cassandra client.
	UseMetrics(metrics any)

	// UseTracer sets the tracer for the Cassandra client.
	UseTracer(tracer any)

	// Connect establishes a connection to Cassandra and registers metrics using the provided configuration when the client was Created.
	Connect()
}
```


Maybe we can open a new issue for this and then make changes to all related driver's methods.